torch/distributed/tensor/experimental/_attention.py

```diff
@@ -104,6 +104,163 @@ class _ContextParallelGlobalVars:
 _cp_global_vars = _ContextParallelGlobalVars()
 
 
+class LoadBalancer(ABC):
+    @abstractmethod
+    def generate_indices(self, restore: bool = False) -> Optional[torch.Tensor]:
+        """
+        Generate indices for load balancing.
+        Args:
+            restore (bool):
+
+        Returns:
+            The generated indices.
+        """
+
+
+class _DummyLoadBalancer(LoadBalancer):
+    def __init__(self) -> None:
+        pass
+
+    def generate_indices(self, restore: bool = False) -> Optional[torch.Tensor]:
```

Comment:
we should figure out how to design this API such that it can also consume the BlockMask or at least parts. I want to  able to build something like https://en.wikipedia.org/wiki/Longest-processing-time-first_scheduling
from the information about the amount of work in the BlockMask